folium/vector_layers.py

```diff
@@ -159,16 +159,27 @@ class PolyLine(Marker):
                     {{this.location}},
                     {{ this.options }}
                     )
-                    {% if this.tooltip %}.bindTooltip("{{this.tooltip.__str__()}}"){% endif %}
+                    {% if this.tooltip %}.bindTooltip("{{ this.tooltip }}"){% endif %}
                     .addTo({{this._parent.get_name()}});
             {% endmacro %}
             """)  # noqa
 
     def __init__(self, locations, popup=None, tooltip=None, **kwargs):
         super(PolyLine, self).__init__(location=locations, popup=popup)
         self._name = 'PolyLine'
-        self.tooltip = tooltip
-
+        if tooltip:
+            if isinstance(tooltip, Tooltip):
+                assert not all((tooltip.text, tooltip.fields)), "Only text " \
```

Comment:
This check is already done in the init of `Tooltip`, don't repeat it.

Reply:
See above.